staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -141,7 +141,8 @@ func createHandler(r rest.NamedCreater, scope *RequestScope, admit admission.Int
 
 		ae := request.AuditEventFrom(ctx)
 		admit = admission.WithAudit(admit, ae)
-		audit.LogRequestObject(ae, obj, scope.Resource, scope.Subresource, scope.Serializer)
+		objGVR := schema.GroupVersionResource{Group: scope.Kind.Group, Version: scope.Kind.Version, Resource: scope.Kind.Kind}
```

Comment:
setting `Resource: scope.Kind.Kind` is definitely not correct.

the group-version-resource passed to LogRequestObject is used for two things which I think need to be made distinct:
1. the coordinates of the objectRef of the request... this should remain as-is (pointing to the scope group/version/resource/subresource)
2. the group-version of the request object, used to encode the object. this should change to use the group-version of the `gvk` returned from `decoder.Decode`

Reply:
@liggitt I agree that the `gv `that object will `encode `comes from `decode`, but scope.kind is also satisfied with `gv`, isn't it?

**Setting gvk to gvr is indeed problematic**, thanks!
I also agree that the `gv `that object will `encode `comes from `decode`, and done.
However,
https://github.com/kubernetes/kubernetes/blob/e6136c0303028d68cac67290d94a60cec167ccdf/staging/src/k8s.io/apiserver/pkg/endpoints/installer.go#L206

https://github.com/kubernetes/kubernetes/blob/e6136c0303028d68cac67290d94a60cec167ccdf/staging/src/k8s.io/apiserver/pkg/endpoints/installer.go#L589

we also can theoretically use` scope.kind.group`, `scope.kind.version` as the `gv `of obj, right?